darwin/DarwinProcess.c

```diff
@@ -240,23 +241,28 @@ void DarwinProcess_setFromKInfoProc(Process* proc, const struct kinfo_proc* ps,
    proc->updated = true;
 }
 
+static double ticksToNanoseconds(const double ticks) {
+   const double nanos_per_sec = 1e9;
+   return ticks / (double) Platform_clockTicksPerSec * Platform_timebaseToNS * nanos_per_sec;
+}
+
 void DarwinProcess_setFromLibprocPidinfo(DarwinProcess* proc, DarwinProcessList* dpl) {
    struct proc_taskinfo pti;
 
    if (sizeof(pti) == proc_pidinfo(proc->super.pid, PROC_PIDTASKINFO, 0, &pti, sizeof(pti))) {
-      if (0 != proc->utime || 0 != proc->stime) {
-         uint64_t diff = (pti.pti_total_system - proc->stime)
-                       + (pti.pti_total_user - proc->utime);
+      double time_interval = ticksToNanoseconds(dpl->global_diff) / (double) dpl->super.cpuCount;
```

Comment:
`double time_interval = ticksToNanoseconds(dpl->global_diff) / (double) dpl->super.cpuCount;` is constant for all processes of a single update cycle.
Maybe pre-compute it in `DarwinProcessList.c::ProcessList_goThroughEntries()`

Reply:
Then this should be passed in as parameter to `DarwinProcess_setFromLibprocPidinfo` to avoid depending on global state.